lib/syskit/runtime/server/write_only_disk_file_system.rb

```diff
@@ -33,15 +36,34 @@ def initialize(data_dir)
                 # If missing, then these commands are not supported.
 
                 def write(ftp_path, stream)
-                    full_path = File.join(@data_dir, ftp_path)
+                    full_path = expand_ftp_path(ftp_path)
                     final_path = File.join(
                         File.dirname(full_path),
                         File.basename(full_path, ".partial")
                     )
                     error "Already exists", 550 if File.exist?(final_path)
 
-                    write_file ftp_path, stream, "wb"
+                    # Code copied from Ftpd::DiskFileSystem::FileWriting
+                    verify_free_space
+                    File.open(full_path, "wb") do |file|
+                        while (line = stream.read)
+                            file.write line
+                            verify_free_space
```

Comment:
```suggestion
                    File.open(full_path, "wb") do |file|
                        while (line = stream.read)
                            verify_free_space_for line.size
                            file.write line
```

and test  `it "does not zero free space if chunk is greater than current free space"`

Reply:
It's very academic (I think we should have thresholds of 10GB or more), but I like the idea nonetheless.

We will need to do some accounting of free space (check everything that takes up space and allocate space as needed).